delira/io/torch.py

```diff
@@ -84,3 +82,85 @@ def load_checkpoint(file, **kwargs):
                     for _key in ["model", "optimizer", "epoch"]]):
             return checkpoint['state_dict']
         return checkpoint
+
+
+    def save_checkpoint_torchscript(file: str, model=None, optimizers={},
+                                    epoch=None, **kwargs):
+        """
+        Save current checkpoint to two different files:
+            1.) ``file + "_model.ptj"``: Will include the state of the model
+                (including the graph; this is the opposite to
+                :func:`save_checkpoint`)
+            2.) ``file + "_trainer_state.pt"``: Will include the states of all
+                optimizers and the current epoch (if given)
+
+        Parameters
+        ----------
+        file : str
+            filepath the model should be saved to
+        model : AbstractPyTorchJITNetwork or None
+            the model which should be saved
+            if None: empty dict will be saved as state dict
+        optimizers : dict
+            dictionary containing all optimizers
+        epoch : int
+            current epoch (will also be pickled)
+
+        """
+
+        # remove file extension if given
+        if any([file.endswith(ext) for ext in [".pth", ".pt", ".ptj"]]):
+            file = file.rsplit(".", 1)[0]
+
+        if isinstance(model, AbstractTorchScriptNetwork):
+            torch.jit.save(model, file + "_model.ptj")
+
+        if optimizers or epoch is not None:
+            save_checkpoint_torch(file + "_trainer_state.pt", None,
+                                  optimizers=optimizers, epoch=epoch, **kwargs)
+
+    def load_checkpoint_torchscript(file: str, **kwargs):
+        """
+        Loads a saved checkpoint consisting of 2 files
+        (see :func:`save_checkpoint_jit` for details)
+
+        Parameters
+        ----------
+        file : str
+            filepath to a file containing a saved model
+        **kwargs:
+            Additional keyword arguments (passed to torch.load)
+            Especially "map_location" is important to change the device the
+            state_dict should be loaded to
+
+        Returns
+        -------
+        OrderedDict
+            checkpoint state_dict
+
+        """
+        # remove file extensions
+        if any([file.endswith(ext) for ext in [".pth", ".pt", ".ptj"]]):
+            file = file.rsplit(".", 1)[0]
+
+        # load model
+        if os.path.isfile(file + ".ptj"):
```

Comment:
I think we should not remove the file extension and only check for the `.ptj` ending. If no file with the `.ptj`ending is found, we should try to load the original file (extension was removed in line 143) and print some message.